views/ListPage/__tests__/ListPage.dataLayer.test.js

```diff
@@ -0,0 +1,67 @@
+import { MockedProvider } from "@apollo/client/testing";
+import { render } from "@testing-library/react-native";
+import React from "react";
+// import renderer from "react-test-renderer";
+import ListPageDataLayer, { GET_LOCATIONS } from "../ListPage.dataLayer";
+
```

Comment:
so one of the reasons that we split our components between datalayers and display layers is for ease of testing. Instead of focusing on mocking out apollo responses and trying to hit every variation here I'd recommend keeping datalayer tests much more simple. 

something like this to ensure that the correct props are being passed down to the display layer. And then on your display layer you can switch between the error state, loading state, and data state in your tests by simply passing in different props. This way minimizes the amount of mocking out external libraries and makes it easier to follow and update tests later on
```
jest.mock('@apollo/client', () => ({
  useQuery: jest.fn(() => ({ })),
}));

describe('ListPageDataLayer', () => {
  it('should render', () => {
    const wrapper = shallow(<ListPageDataLayer/>);
    expect(wrapper).toMatchSnapshot();
  });
});

```